CodeGen/Generators/UnitsNetGen/QuantityGenerator.cs

```diff
@@ -517,7 +517,7 @@ private void GenerateStaticParseMethods()
         /// <example>
         ///     Length.Parse(""5.5 m"", CultureInfo.GetCultureInfo(""en-US""));
         /// </example>
-        public static bool TryParse(string? str, out {_quantity.Name} result)
+        public static bool TryParse([NotNullWhen(true)]string? str, out {_quantity.Name} result)
```

Comment:
This is of course not necessary (and would probably increase our file size a little) but this is how the `IParsable` interface has it...
Haven't tried it, but I image we would get no warning in the block below:

``` csharp

if (Mass.TryParse(nullableStringAsInput, out _)){
  var firstChar = nullableStringAsInput[0]; // I guess this should be safe access..
}
```

Reply:
We should definitely do this, and seeing how widely used this is I don't imagine it increases any binary size in practice - meaning, the nuget is probably already loaded if not already bundled in the target framework.

Reply:
Okay, maybe a slight extra overhead in metadata of our methods, but still - worth it.